skycoin/skywire
# Fix app startup issues after changing the args via UI

Did you run `make format && make check`?
Yes
Fixes #627 

 Changes:	
- Args to the app binary are now taken from the config, not from the previous app start
- App proc is now correctly removed from the proc manager on app exit, no matter what was the exit reason

How to test this PR:
To test that hypervisor restarts app with the changed args:
1. Start any app via UI, VPN client for example
2. Change PK via UI
3. VPN client should be restarted and connected to the new remote server

To test the original issue:
1. Modify `./internal/vpn/client.go`: put this line `return errors.New("fail")` to the line around 183rd, right after the `dialServeConn` and checking its error. This will make the app exit right after connection failure
2. Build binaries
3. Modify VPN client config to connect to the remote visor which doesn't have VPN server 
4. Run VPN client from the UI. After a while it will be displayed as being off again, wait for it
5. Change the remote PK to the visor which has VPN enabled. It should change PK okay
6. Run VPN client from the UI, you should be using VPN now okay
